internal/archive/archive_test.go

```diff
@@ -100,7 +100,7 @@ func testArchive(t *testing.T, when spec.G, it spec.S) {
 
 				tw := tar.NewWriter(fh)
 
-				err = archive.WriteDirToTar(tw, src, "/nested/dir/dir-in-archive", 1234, 2345, 0777, true)
+				err = archive.WriteDirToTar(tw, src, "/nested/dir/dir-in-archive", 1234, 2345, 0777, true, nil)
```

Comment:
It's starting to look like we're got a decent amount of options here. Specially for optional args, we should probably move to  using `With` options pattern.

Reply:
We can opt to punt on this but I can see that only the first 3 args are required and all others should use `With` ops pattern and have sensible defaults.

Reply:
Sounds good.  Originally, I was looking for a way to perform method overloading, but unfortunately go doesn't support this.  This pattern seems like a good alternative.  I'll punt on this for this PR, but will include it into a separate refactor PR.